internal/contract/controlplane.go

```diff
@@ -226,3 +226,10 @@ func (c *ControlPlaneMachineTemplate) NodeDrainTimeout() *Duration {
 		path: Path{"spec", "machineTemplate", "nodeDrainTimeout"},
 	}
 }
+
+// NodeDeletionTimeout provides access to the nodeDeletionTimeout of a MachineTemplate.
+func (c *ControlPlaneMachineTemplate) NodeDeletionTimeout() *Duration {
+	return &Duration{
+		path: Path{"spec", "machineTemplate", "nodeDeletionTimeout"},
+	}
+}
```

Comment:
What solved it for me was the following func for Set (in `internal/contract/types` 

```go
func (i *Duration) Set(obj *unstructured.Unstructured, value metav1.Duration) error {
	durationString, err := value.MarshalJSON()
	if err != nil {
		return errors.Wrapf(err, "failed to marshal duration %s", value.Duration.String())
	}
	unquotedString, err := strconv.Unquote(string(durationString))
	if err != nil {
		return errors.Wrapf(err, "failed to marshal duration %s", value.Duration.String())
	}
	if err := unstructured.SetNestedField(obj.UnstructuredContent(), unquotedString, i.path...); err != nil {
		return errors.Wrapf(err, "failed to set path %s of object %v", "."+strings.Join(i.path, "."), obj.GroupVersionKind())
	}
	return nil
}

```

The issue seemed to be that the string ends up being double quoted and the API server can't interpret it. I think we should at least add these values to our integration tests (which include webhooks) and maybe the e2e tests so we can be sure they're settable.

Reply:
Thank you for help figure this out, I see that the similar changes where done for NodeDrainTimeout via the below the commit
https://github.com/kubernetes-sigs/cluster-api/commit/819662ccb3911e38672a3a030f38c6183608e4e7
I will make the changes too and push it